internal/cluster/distribution/eks/eksprovider/workflow/create_addon_activity.go

```diff
@@ -0,0 +1,170 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"go.uber.org/cadence/activity"
+
+	"github.com/banzaicloud/pipeline/internal/cluster/infrastructure/aws/awsworkflow"
+)
+
+const CreateAddonActivityName = "eks-create-addon"
+
+// CreateAddonActivity creates an EKS addon
+type CreateAddonActivity struct {
+	awsSessionFactory *awsworkflow.AWSSessionFactory
+}
+
+// CreateAddonActivityInput holds input data
+type CreateAddonActivityInput struct {
+	EKSActivityInput
+
+	KubernetesVersion string
+	AddonName         string
+}
+
+// CreateAddonActivityOutput holds the output data
+type CreateAddonActivityOutput struct {
+}
+
+// NewCreateAddonActivity instantiates a new CreateAddonActivity
+func NewCreateAddonActivity(awsSessionFactory *awsworkflow.AWSSessionFactory) *CreateAddonActivity {
+	return &CreateAddonActivity{
+		awsSessionFactory: awsSessionFactory,
+	}
+}
+
+func (a *CreateAddonActivity) Execute(ctx context.Context, input CreateAddonActivityInput) (*CreateAddonActivityOutput, error) {
+	logger := activity.GetLogger(ctx).Sugar().With(
+		"organization", input.OrganizationID,
+		"cluster", input.ClusterName,
+		"region", input.Region,
+		"version", input.KubernetesVersion,
+	)
+
+	awsSession, err := a.awsSessionFactory.New(input.OrganizationID, input.SecretID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil {
+		return nil, err
+	}
+	eksSvc := eks.New(
+		awsSession,
+		aws.NewConfig().
+			WithLogger(aws.LoggerFunc(
+				func(args ...interface{}) {
+					logger.Debug(args)
+				})).
+			WithLogLevel(aws.LogDebugWithHTTPBody),
+	)
+
+	logger = logger.With("addonName", input.AddonName)
+	logger.Info("create add-on for cluster : " + (input.ClusterName))
+	t := time.Now().Unix()
+
+	addOnInput := &eks.CreateAddonInput{
+		AddonName:        aws.String(input.AddonName),
+		ClusterName:      aws.String(input.ClusterName),
+		ResolveConflicts: aws.String(eks.ResolveConflictsOverwrite),
+	}
+	addOnOutput, err := eksSvc.CreateAddon(addOnInput)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceInUseException" {
+			logger.Info("addon created but: " + awsErr.Message())
```

Comment:
May we can log at warn level, I accidentally run into this error during development sometimes, that the create addon request has failed with this ResourceInUseException error, but right after that when making the the describe request to get it's status returned a valid status and later installed successfully. In case addon install ultimately fails we will return with an error anyway with waitUntilAddOnCreateCompleteWithContext.